typing/ctype.ml

```diff
@@ -730,8 +727,10 @@ let duplicate_class_type ty =
 *)
 let rec generalize ty =
   let ty = repr ty in
+  (* generalize the type iff ty.level <= !current_level *)
   if (ty.level > !current_level) && (ty.level <> generic_level) then begin
     set_level ty generic_level;
+    (* recur into abbrev for the speed *)
```

Comment:
Also, I think this comment (apart from being obscure to me) is at odds with the comment above the function.
So, which is it?